framework/include/problems/FEProblemBase.h

```diff
@@ -2348,6 +2348,9 @@ class FEProblemBase : public SubProblem, public Restartable
   virtual void setCurrentLowerDElem(const Elem * const lower_d_elem, const THREAD_ID tid) override;
   virtual void setCurrentBoundaryID(BoundaryID bid, const THREAD_ID tid) override;
 
+  // Used for setting previous states
+  int _global_current_state;
```

Comment:
Is this public? It's kinda difficult to check in the web browser, but if so move it to protected at least.